tests/test_string_parsing.py

```diff
@@ -0,0 +1,46 @@
+'''
+Check if the string parsing works as expected
+'''
+import pytest
+from string_parser.string_parser import StringParser
+
+@pytest.mark.parametrize("test_input,expected", [(pytest.string_with_spaces,
+                                                  {'Variables': 1, 'may': 1, 'subsequently': 1,
+                                                   'be': 1, 'rebound': 1, 'at': 1,
+                                                   'any': 2, 'time': 1, 'to': 1,
+                                                   'object': 1}),
+                                                 (pytest.string_with_comas_dots,
+                                                  {'It': 1, 'has': 1, 'list': 1,
+                                                   'comprehensions': 1, 'and': 2,
+                                                   'dictionaries': 1, 'Sets': 1, 'generator': 1,
+                                                   'expressions': 1}),
+                                                 (pytest.string_with_dashes,
+                                                  {'Python': 1, 'is': 2, 'a': 1,
+                                                   'high-level': 1, 'general-purpose': 1,
+                                                   'programming': 1, 'language': 1,
+                                                   'which': 1, 'beginner-oriented': 1}),
+                                                 (pytest.string_quote_parentheses_new_line,
+                                                  {"Python's": 1, 'development': 1, 'is': 1,
+                                                   'conducted': 1, 'largely': 1, 'through': 1,
+                                                   'the': 2, 'Python': 1, 'Enhancement': 1,
+                                                   'Proposal': 1, 'PEP': 1, 'process': 1,
+                                                   'primary': 1, 'mechanism': 1, 'for': 1,
+                                                   'proposing': 1, 'major': 1, 'features': 1})])
```

Comment:
This is hardly readable.
```suggestion
@pytest.mark.parametrize(
    "test_input,expected", 
    [(pytest.string_with_spaces,
     {'Variables': 1, 'may': 1, 'subsequently': 1, 'be': 1, 'rebound': 1, 'at': 1, 'any': 2, 'time': 1, 'to': 1, 'object': 1}),
     (pytest.string_with_comas_dots,
     {'It': 1, 'has': 1, 'list': 1, 'comprehensions': 1, 'and': 2, 'dictionaries': 1, 'Sets': 1, 'generator': 1, 'expressions': 1}),
...
```
And so on.

Or we can move the test data along with expected results to separate variables. So here you just only specify the varaibles